src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/global.yml

```diff
@@ -1244,9 +1274,70 @@ Resources:
             - master
       Targets:
         - Arn:
-            !Join [ '', [ 'arn:aws:codepipeline:', !Ref 'AWS::Region', ':', !Ref 'AWS::AccountId', ':', !Ref CodePipeline ] ]
+            !Join [
+              "",
+              [
+                "arn:aws:codepipeline:",
+                !Ref "AWS::Region",
+                ":",
+                !Ref "AWS::AccountId",
+                ":",
+                !Ref CodePipeline,
+              ],
+            ]
           RoleArn: !GetAtt PipelineCloudWatchEventRole.Arn
           Id: adf-codepipeline-trigger-pipeline
+  TerraformLockTable:
+    Type: "AWS::DynamoDB::Table"
+    Properties:
+      AttributeDefinitions:
+        - AttributeName: LockID
+          AttributeType: S
+      KeySchema:
+        - AttributeName: LockID
+          KeyType: HASH
+      ProvisionedThroughput:
+        ReadCapacityUnits: 5
+        WriteCapacityUnits: 5
+      TableName: adf-tflocktable
+  ADFTerraformRole:
+    # ADF role to run terraform pipelines
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: "adf-terraform-role"
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Sid: "AssumeRole"
+            Principal:
+              AWS:
+                # This would allow all codebuild projects to be able to assume this role
+                - !Sub arn:aws:iam::${AWS::AccountId}:role/adf-codebuild-role
+                # - !Sub arn:aws:iam::${DeploymentAccountId}:role/my-custom-codebuild-role
```

Comment:
This suggestion implies that users could think they need to change this file.
While it would be better to change the global-iam.yml instead.

Maybe all of these resources need to go to the example global iam file, commented out and then the users that would like to use Terraform can enable it by copying that into their global-iam.yml file instead.